tests/integration/thrift_tests.py

```diff
@@ -224,6 +227,31 @@ def example(self, context):
         self.assertEqual(handler.server_span.sampled, False)
         self.assertTrue(client_result)
 
+    def test_budget_header(self):
+        """Test that the budget header is set in the headers if the client sets it."""
+        budget = "1234"
+
+        class Handler(TestService.Iface):
+            def __init__(self):
+                self.server_span = None
+
+            def example(self, context):
+                self.server_span = context.span
+                self.context = context
+                return True
+
+        handler = Handler()
+
+        with serve_thrift(handler, TestService) as server:
+            with raw_thrift_client(server.endpoint, TestService) as client:
+                transport = client._oprot.trans
+                transport.set_header(b"Deadline-Budget", budget.encode())
+                client_result = client.example()
+
+        self.assertIsNotNone(handler.server_span)
```

Comment:
nit: since this test is called `test_budget_header`, this assert sounds unrelated.